src/transactions.rs

```diff
@@ -205,37 +225,50 @@ impl<'a, L: Ledger + Serialize + DeserializeOwned> TransactionEditor<'a, L> {
     }
 }
 
-impl<'a, L: Ledger + Serialize + DeserializeOwned> Deref for TransactionEditor<'a, L> {
+impl<'a, L: Ledger> Deref for TransactionEditor<'a, L> {
     type Target = Transaction<L>;
 
     fn deref(&self) -> &Transaction<L> {
         &self.transaction
     }
 }
 
-impl<'a, L: Ledger + Serialize + DeserializeOwned> DerefMut for TransactionEditor<'a, L> {
+impl<'a, L: Ledger> DerefMut for TransactionEditor<'a, L> {
     fn deref_mut(&mut self) -> &mut Transaction<L> {
         &mut self.transaction
     }
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(bound = "")]
 pub struct TransactionParams<L: Ledger> {
-    pub uid: TransactionUID<L>,
+    pub uid: Option<TransactionUID<L>>,
     pub timeout: Option<u64>,
     pub status: TransactionStatus,
     pub memos: Vec<Option<ReceiverMemo>>,
-    pub sig: Signature,
+    pub sig: Option<Signature>,
     pub inputs: Vec<RecordOpening>,
     pub outputs: Vec<RecordOpening>,
     pub time: DateTime<Local>,
     pub asset: AssetCode,
     pub kind: TransactionKind<L>,
     pub senders: Vec<UserAddress>,
     pub receivers: Vec<(UserAddress, RecordAmount)>,
+    pub fee_change: Option<RecordAmount>,
+    pub asset_change: Option<RecordAmount>,
+}
+
+impl<L: Ledger> PartialEq<Self> for TransactionParams<L> {
+    fn eq(&self, other: &Self) -> bool {
+        self.uid == other.uid
+            && self.kind == other.kind
+            && self.timeout == other.timeout
+            && self.time == other.time
```

Comment:
This seems incomplete. Is it possible to use `derivative` to derive `PartialEq` so we don't have to worry about updating this whenever we add a field?

Reply:
I will try this!